workflow/workflow_test.go

```diff
@@ -306,7 +307,8 @@ func TestWithDoubleSerializationDetection(t *testing.T) {
 			map[string]any{},
 		},
 	}
-	for _, i := range testIter {
+	for i, testData := range testIter {
+		t.Logf("Starting iteration %d", i)
```

Comment:
Inserting log messages into tests can be really useful for debugging ... do you really mean to merge these into `main`?? Chatty test output can often be unnecessarily difficult to read ...